imap_processing/ultra/l2/ultra_l2.py

```diff
@@ -0,0 +1,334 @@
+                "Please ensure the dataset contains all required variables."
+            )
+    return ultra_pointing_set
+
+
+def generate_ultra_healpix_skymap(
+    ultra_l1c_psets: list[str | xr.Dataset],
+    output_map_properties: MapProperties = DEFAULT_L2_MAP_PROPERTIES,
+) -> ena_maps.HealpixSkyMap:
+    """
+    Generate a Healpix skymap from ULTRA L1C pointing sets.
+
+    This function combines IMAP Ultra L1C pointing sets into a single L2 HealpixSkyMap.
+    It handles the projection of values from pointing sets to the map,applies necessary
+    weighting and background subtraction, and calculates flux and flux uncertainty.
+
+    Parameters
+    ----------
+    ultra_l1c_psets : list[str | xr.Dataset]
+        List of paths to ULTRA L1C pointing set files or xarray Datasets containing
+        pointing set data.
+    output_map_properties : MapProperties, optional
+        Properties defining the output map configuration. If not provided, default L2
+        map properties will be used.
+
+    Returns
+    -------
+    ena_maps.HealpixSkyMap
+        HealpixSkyMap object containing the combined data from all pointing sets,
+        with calculated flux and flux uncertainty values.
+
+    Notes
+    -----
+    The structure of this function goes as follows:
+    1. Initialize the HealpixSkyMap object with the specified properties.
+    2. Iterate over the input pointing sets and read them into UltraPointingSet objects.
+    3. For each pointing set, weight certain variables by exposure and solid angle of
+    the pointing set pixels.
+    4. Project the pointing set values to the map using the push method.
+    5. Perform subsequent processing for weighted quantities at the SkyMap level
+    (e.g., divide weighted quantities by their summed weights to
+    get their weighted mean)
+    6. Calculate corrected count rate with background subtraction applied.
+    7. Calculate flux and flux uncertainty.
+    8. Drop unnecessary variables from the map.
+    """
+    if output_map_properties.sky_tiling_type is ena_maps.SkyTilingType.HEALPIX:
+        map_nside, map_nested = (
+            output_map_properties.nside,
+            output_map_properties.nested,
+        )
+    else:
+        map_nside, map_nested = (DEFAULT_L2_HEALPIX_NSIDE, DEFAULT_L2_HEALPIX_NESTED)
+
+    # Initialize the HealpixSkyMap object
+    skymap = ena_maps.HealpixSkyMap(
+        nside=map_nside,
+        nested=map_nested,
+        spice_frame=output_map_properties.spice_reference_frame,
+    )
+
+    # Add additional data variables to the map
+    output_map_properties.values_to_push_project.extend(
+        [
+            "observation_time",
+            "pointing_set_exposure_times_solid_angle",
+            "num_pointing_set_pixel_members",
+        ]
+    )
+
+    for ultra_l1c_pset in ultra_l1c_psets:
+        pointing_set = read_into_pointing_set(ultra_l1c_pset)
+        logger.info(f"PSET epoch: {pointing_set.epoch}")
```

Comment:
Nice logging! Maybe add some more info / quantities about what you're projecting from this pset too?

Reply:
```
        logger.info(
            f"Projecting a PointingSet with {pointing_set.num_points} pixels "
            f"at epoch:{pointing_set.epoch}\n"
            f"These values will be projected: {value_keys_to_push_project}"
        )
```

Reply:
Add the above, thanks!